pkg/controller/common/certificates/http_reconcile.go

```diff
@@ -216,29 +217,45 @@ func ensureInternalSelfSignedCertificateSecretContents(
 			owner, namer, tls, controllerSANs, svcs, parsedCSR, rotationParam.Validity,
 		)
 		// sign the certificate
-		certData, err := ca.CreateCertificate(*validatedCertificateTemplate)
+		certificate, err = ca.CreateCertificate(*validatedCertificateTemplate)
 		if err != nil {
 			return secretWasChanged, err
 		}
 
 		secretWasChanged = true
 		// store certificate and signed certificate in a secret mounted into the pod
 		secret.Data[CAFileName] = EncodePEMCert(ca.Cert.Raw)
-		secret.Data[CertFileName] = EncodePEMCert(certData, ca.Cert.Raw)
+		secret.Data[CertFileName] = EncodePEMCert(certificate, ca.Cert.Raw)
```

Comment:
I wonder if we should avoid the redundancy of calling `EncodePEMCert` twice for both the CA and the certificate here and on L232-233

Reply:
The idea was to have 2 different log messages according to what is generated/updated:
1. A new certificate is generated: "Issuing new HTTP certificate" is logged
2. No new certificate is generated, only the CA should be updated: "Updating CA certificate" is logged

I can remove the redundancy, and only call `EncodePEMCert` on L232-233, but it means that I should remove the "Updating CA certificate" message since it is no more accurate (it would also be logged if the certificate has been updated, not only the CA).

Reply:
I see the dilemma. I think the log message is valuable. I am sure we could fine a way but it would probably be harder to read so let's not make this micro optimisation for now.